docker-compose.yaml

```diff
@@ -0,0 +1,83 @@
+version: '2'
+
+services:
+
+  main:
+    build:
+      context: .
+      dockerfile: Dockerfile
+      args:
+        - http_proxy
+        - https_proxy
+        - no_proxy
+        - BUILDDOC=OFF
+        - BUILDSWIG=OFF
+        - BUILDSWIGPYTHON=OFF
+        - BUILDSWIGJAVA=OFF
+        - BUILDSWIGNODE=OFF
+        - USBPLAT=OFF
+        - FIRMATA=OFF
+        - ONEWIRE=OFF
+        - JSONPLAT
+        - IMRAA=OFF
+        - FTDI4222=OFF
+        - IPK=OFF
+        - RPM=OFF
+        - ENABLEEXAMPLES=ON
+        - INSTALLGPIOTOOL=OFF
+        - INSTALLTOOLS=OFF
+        - BUILDTESTS=ON
+        - CC=clang-3.8
+        - CXX=clang++-3.8
+        - NODE_VERSION=v4.1.2
+        - BUILDARCH
+    environment:
+      - http_proxy
+      - https_proxy
+      - no_proxy
+
+  doc:
+    extends: main
+    build:
+      args:
+        - BUILDDOC=ON
+    command: make doc
+    volumes:
+      - ./build/html:/usr/src/app/build/html
+
+  python2:
+    extends: main
+    build:
+      args:
+        - BUILDSWIG=ON
+        - BUILDSWIGPYTHON=ON
+    command: bash -c "make _python2-mraa && make test"
+
+  python3:
+    extends: python2
+    command: bash -c "make _python3-mraa && make test"
+
+  java:
+    extends: main
+    build:
+      args:
+        - BUILDSWIG=ON
+        - BUILDSWIGJAVA=ON
+        - CC=gcc-4.8
+        - CXX=g++-4.8
+    command: bash -c "make mraajava && make test"
+
+  node4:
+    extends: main
+    build:
+      args:
+        - BUILDSWIG=ON
+        - BUILDSWIGNODE=ON
+        - NODE_VERSION=v4.1.2
```

Comment:
Sounds good to me! I actually choose this version because it was being used in the "old" travis file. It can be easily changed